airflow/sensors/base.py

```diff
@@ -322,7 +322,7 @@ def run_duration() -> float:
                 raise AirflowRescheduleException(reschedule_date)
             else:
                 time.sleep(self._get_next_poke_interval(started_at, run_duration, poke_count))
-                poke_count += 1
+            poke_count += 1
```

Comment:
Do you test that this changes are actually works? 

When task rescheduled it would create TaskInstance and initialise new object that mean that there is no access to variables from the previous run.

Reply:
Ohh correct, sorry missed it. Yes this wont work as TaskInstance object is recreated at end of schedule. This means to generate next_poke_interval we need to store the poke_count (or try_number in execute function) somewhere externally, or infer it from existing information.

I also came across #30669 , i am not sure what would be the next steps to address this bug :|
PR #30669 pretty much handles this bug

Reply:
> or infer it from existing information.

I think this should be the solution to this problem, we remove the poke_count (or try_number) from the execute function altogether, and estimate the try number based on available information such as start_date and run_duration (or duration from TaskReshedule). If someone can review this logic and give feedback that would be helpful. 

for now ill push a change to this PR as a proposed solution